src/components/NotificationAlert.tsx

```diff
@@ -0,0 +1,33 @@
+import { ReactNode } from 'react';
+import { Alert, AlertTitle } from '@mui/material';
+import { useNotificationContext } from '../providers/NotificationProvider.tsx';
+
+interface NotificationAlertProps {
+  severity: 'success' | 'error' | 'info' | 'warning';
+  title?: string;
+  children: ReactNode;
+}
+
+export function NotificationAlert({
```

Comment:
You can render this component directly in the NotificationProvider only once, and even omit the `showNotification` value in the context. Then it becomes something like

```ts
function NotificationProvider({ children }: { children: ReactNode }) {
  const [{ show, title }, setShowNotification] = useState({ show: false, title: string });

  const value = {
    setShowNotification,
  };

  return (
    <NotificationContext.Provider value={value}>
     {show && <NotificationAlert title={title} />}
      {children}
    </NotificationContext.Provider>
  );
}
```

And you don't need to render the NotificationAlert anywhere else. Just call setShowNotification({ show: true, title: 'Something went wrong' }) wherever it's needed  


```